miner/worker.go

```diff
@@ -1196,8 +1196,42 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 		if w.commitBundle(bundleTxs, w.coinbase, interrupt) {
 			return
 		}
-		w.current.profit.Add(w.current.profit, bundle.totalEth)
+		w.current.profit.Add(w.current.profit, bundle.ethSentToCoinbase)
 	}
+	if w.flashbots.isMegabundleWorker {
+		megabundle, err := w.eth.TxPool().GetMegabundle(w.flashbots.relayAddr, header.Number, header.Time)
+		log.Info("Starting to process a Megabundle", "relay", w.flashbots.relayAddr, "megabundle", megabundle, "error", err)
+		if err != nil {
+			return // no valid megabundle for this relay, nothing to do
+		}
+		// Flashbots bundle merging duplicates work by simulating TXes and then committing them once more.
+		// Megabundles API focuses on speed and runs everything in one cycle.
+		coinbaseBalanceBefore := w.current.state.GetBalance(w.coinbase)
+		if w.commitBundle(megabundle.Txs, w.coinbase, interrupt) {
```

Comment:
I find it slightly confusing that `commitBundle` returns `true` if it has an error. Would it be better to return an `error`?

Reply:
It's intended to follow the same pattern as `commitTransactions`, although I agree it's non-intuitive